packages/scripts/src/monitor-dvm/MonitorDeletion.ts

```diff
@@ -0,0 +1,25 @@
+import { Logger } from "@uma/financial-templates-lib";
+import { VotingV2Ethers } from "@uma/contracts-node";
+import { logDeletionProposed } from "./MonitorLogger";
+import { getContractInstanceByUrl } from "../utils/contracts";
+import type { MonitoringParams } from "./common";
+
+export async function monitorDeletion(logger: typeof Logger, params: MonitoringParams): Promise<void> {
+  const votingV2 = await getContractInstanceByUrl<VotingV2Ethers>("VotingV2", params.jsonRpcUrl);
+
+  const deletionProposals = (
+    await votingV2.queryFilter(
+      votingV2.filters.SignaledRequestsAsSpamForDeletion(),
+      params.startingBlock,
+      params.endingBlock
+    )
+  ).map((event) => ({
+    tx: event.transactionHash,
+    proposalId: event.args.proposalId.toString(),
+    sender: event.args.sender,
+    spamRequestIndices: event.args.spamRequestIndices,
+  }));
+  deletionProposals.forEach((proposal) => {
+    logDeletionProposed(logger, proposal, params.chainId);
```

Comment:
small nit but not sure why you have the `.map`, you could just change it to a `.forEach` and log with one iteration and less code, alternatively you could chain the map into the for each without needing to store the extra variable `deletionProposals`

Reply:
I was not able to reduce lines of code as significant part of this deals with passing `proposal` parameters to `logDeletionProposed`.